### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by g41797**
Tests log on the local computer:
```bash
~/go/pkg/mod/github.com/memphisdev/memphis.go$ go test -v
=== RUN   TestConnect
--- PASS: TestConnect (1.02s)
=== RUN   TestNormalizeHost
--- PASS: TestNormalizeHost (0.00s)
=== RUN   TestProduceNoProducer
--- PASS: TestProduceNoProducer (1.02s)
=== RUN   TestCreateProducer
--- PASS: TestCreateProducer (1.16s)
=== RUN   TestProduce
--- PASS: TestProduce (1.09s)
=== RUN   TestRemoveProducer
--- PASS: TestRemoveProducer (1.10s)
=== RUN   TestFetch
2023/11/26 12:03:14 Consumer consumer_a: station unreachable
2023/11/26 12:03:14 Consumer consumer_a: consumer is inactive
--- PASS: TestFetch (1.11s)
=== RUN   TestConsume
--- PASS: TestConsume (1.10s)
=== RUN   TestCreateConsumer
--- PASS: TestCreateConsumer (1.11s)
=== RUN   TestRemoveConsumer
--- PASS: TestRemoveConsumer (1.11s)
=== RUN   TestFullFlow
--- PASS: TestFullFlow (0.18s)
=== RUN   TestCreateSchema
sdk_test_schema_graphql already exist in the db
sdk_test_schema_protobuf2 already exist in the db
sdk_test_schema_protobuf3 already exist in the db
sdk_test_schema_json already exist in the db
sdk_test_schema_avro already exist in the db
--- PASS: TestCreateSchema (1.02s)
=== RUN   TestCreateStation
--- PASS: TestCreateStation (1.22s)
=== RUN   TestRemoveStation
--- PASS: TestRemoveStation (1.19s)
=== RUN   TestCreateStationWithDefaults
--- PASS: TestCreateStationWithDefaults (1.08s)
PASS
ok      github.com/memphisdev/memphis.go        16.109s
```
Docker images automatically downloaded and composed by test.
Url of docker-compose.yml taken from [dev.env](https://github.com/memphisdev/memphis.go/blob/master/_configuration/dev.env) file:
```bash
COMPOSEURL=https://memphisdev.github.io/memphis-docker/docker-compose-dev.yml
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/memphisdev/memphis.go
 
-go 1.18
+go 1.19
```
Comment (idanasulinmemphis) [inline go.mod:3]: We've upgraded to 1.20 in the last couple of days

File: Jenkinsfile [hunk 2]
```diff
@@ -13,8 +13,8 @@ node ("small-ec2-fleet") {
   
   try{
     stage ('Install GoLang') {
-      sh 'wget -q https://go.dev/dl/go1.18.4.linux-amd64.tar.gz'
-      sh 'sudo  tar -C /usr/local -xzf go1.18.4.linux-amd64.tar.gz'
+      sh 'wget -q https://go.dev/dl/go1.19.13.linux-amd64.tar.gz'
```
Comment (idanasulinmemphis) [inline Jenkinsfile:16]: This file will be edited by us so please remove this change

File: README.md [hunk 3]
```diff
@@ -827,3 +827,6 @@ consumer.Destroy();
 ```go
 conn.IsConnected()
 ```
+
+# Test status [![Test](https://github.com/memphisdev/memphis.go/actions/workflows/test.yml/badge.svg)](https://github.com/memphisdev/memphis.go/actions/workflows/test.yml)
```
Comment (idanasulinmemphis) [inline README.md:831]: remove this section as well please

File: go.mod [hunk 4]
```diff
@@ -9,13 +9,18 @@ require (
 )
 
 require (
+	github.com/g41797/sputnik v0.0.21
```
Comment (idanasulinmemphis) [inline go.mod:12]: we can not rely on this package since it has not wide community support and maintenance

File: producer_test.go [hunk 5]
```diff
@@ -25,17 +25,8 @@ func TestCreateProducer(t *testing.T) {
...
-	}
-
-	_, err = s.CreateProducer("producer_name_a", ProducerGenUniqueSuffix())
-	if err != nil {
-		t.Error(err)
-	}
-	_, err = s.CreateProducer("producer_name_a", ProducerGenUniqueSuffix())
 	if err != nil {
-		t.Error(err)
+		t.Error("Producer names have not to be unique")
```
Comment (idanasulinmemphis) [inline producer_test.go:29]: just log the error

File: producer_test.go [hunk 6]
```diff
@@ -44,8 +35,8 @@ func TestCreateProducer(t *testing.T) {
 	}
 
 	_, err = c.CreateProducer("station_name_1", "producer_name_b")
-	if err == nil {
-		t.Error("Producer names has to be unique")
+	if err != nil {
+		t.Error("Producer names have not to be unique")
```
Comment (idanasulinmemphis) [inline producer_test.go:39]: just log the error

File: station.go [hunk 7]
```diff
@@ -462,7 +462,7 @@ func (c *Conn) removeFunctionsUpdatesListener(stationName string) error {
 
 	sfs, ok := c.stationFunctionSubs[sn]
 	if !ok {
-		return memphisError(errors.New("functions listener doesn't exist"))
+		return nil
```
Comment (idanasulinmemphis) [inline station.go:465]: all changes in this file seem like an issue but you return nil, could you please explain?

**Review by idanasulinmemphis — CHANGES_REQUESTED** [review]
Hi @g41797 I left some comments, please fix them.
Thanks for that

**Comment by g41797** [comment]
get rid of [github.com/g41797/sputnik](http://github.com/g41797/sputnik) is problematic, because _updated testing flow_ is based on this package
it's small problem - I can simply close PR

for protocol-adapter it's another story - _**whole functionality of adapter  is based on sputnik**_

**Pull request closed without merging**